src/main/resources/templates/db_scripts/updateItemStatusDate.sql

```diff
@@ -11,8 +11,13 @@ AS $$
 	    -- Date time in "YYYY-MM-DD"T"HH24:MI:SS.ms'Z'" format at UTC (00:00) time zone
       NEW.jsonb = jsonb_set(NEW.jsonb, '{status,date}',
        to_jsonb(to_char(CURRENT_TIMESTAMP(3) AT TIME ZONE 'UTC', 'YYYY-MM-DD"T"HH24:MI:SS.ms"Z"')), true);
+    ELSIF (OLD.jsonb->'status'->'date' IS NOT NULL) THEN
+      NEW.jsonb = jsonb_set(NEW.jsonb, '{status,date}', OLD.jsonb->'status'->'date', true);
+    ELSE
```

Comment:
@bohdan-suprun Could you please describe - why did you add this line:  `NEW.jsonb = NEW.jsonb` ?

Reply:
Every `readonly` field is being removed by RMB when receiving a request. This prevents a readonly field to be changed. But another side of this behavior that it is being removed regardless whether it has changed or not.

Because of this we have to restore the value of the field every time we do an update.

This code sets the `NEW.jsonb->'status'->'date'` to `OLD.jsonb->'status'->'date'` if previous date is not null. And the `ELSE` block it is additional logic required to temporary fix https://issues.folio.org/browse/RMB-538.

Reply:
Oh, I catch your point. Thanks!